### ISSUE TITLE
Emacs FSI hangs on evaluting expression ending without double semicolon

### ISSUE THREAD

**Issue by delexi** [fsharp/fsharpbinding#973]
M-x version 
`GNU Emacs 24.4.1 (x86_64-unknown-linux-gnu, GTK+ Version 3.14.7) of 2015-01-17 on bisson`

fsharp-mode
`Version 1.5.1 installed from Melpa Stable`

C-h v inferior-fsharp-program
`"fsharpi --readline-"`

I press C-c C-s in a .fs file (which switches to the `*inferior-fsharp*` buffer) and enter an expression without a double semicolon at the end. After pressing RET emacs hangs. I can interrupt the hanging by pressing C-g. After this, a newline is inserted after the expression in the `*inferior-fsharp*` buffer. However, no prompt ">" is inserted. I can now enter the ";;", press RET and the expression evaluates as expected.

**Comment by rneatherway** [comment]
Hi @delexi, thanks for reporting. There are a few things I don't understand about your issue though. First, `C-c C-s` shouldn't switch to the `*inferior-fsharp*` buffer and doesn't for me. Is this when you first launch FSI? Second, when I type `let x = 1 <RET>` at the prompt emacs doesn't hang -- I can press enter a few more times and then enter `;;` when I like. This behaviour is just the way that FSI works.

**Comment by delexi** [comment]
This is what happens, when I press `C-c C-s` in an `fsharp-mode`-buffer (when no `*inferior-fsharp*`-buffer is open:

1. I am prompted for the interpreter command to run (with the value of `inferior-fsharp-program` as the default): `fsharp toplevel to run: /usr/bin/fsharpi --readline-`

2. I press `RET` which opens the `*inferior-fsharp*` buffer in another window:
  ```
  F# Interactive for F# 3.1 (Open Source Edition)
  Freely distributed under the Apache 2.0 Open Source License
  
  For help type #help;;
  
  > 
  ```
3. I switch to said buffer and enter `let x = 1 RET`, emacs hangs as described above.

I know how the fsharp REPL usually works - if I start it in a terminal-emulator (e.g. `ansi-term`) it behaves as you describe.

As for `C-c C-s`: This is bound to `fsharp-show-subshell`.

**Comment by rneatherway** [comment]
I've just fired up emacs with no `init.el` in a clean `HOME` directory, installed `fsharp-mode` and tried to reproduce, but it still works fine. My version is `GNU Emacs 24.4.1 (x86_64-pc-linux-gnu, GTK+ Version 3.14.5) of 2014-12-09 on gaia, modified by Debian`.

Could you try in a clean `HOME`? I just did:

    mkdir tmp
    cd tmp
    HOME=`pwd` emacs

and then added MELPA stable as a package source and installed `fsharp-mode`.

**Comment by delexi** [comment]
It works perfectly with a clean emacs. So this has something todo with one of my installed packages or some settings in my `init.el`. Do you have any idea, what this could be?

**Comment by rneatherway** [comment]
I see, glad that is resolved. I would be happy to accept a PR that locally forced this variable to `nil` for inferior fsharpi.

**Comment by rneatherway** [comment]
Fixed by #978. Thanks @delexi
